Code/Framework/AzCore/Tests/Name/NameTests.cpp

```diff
@@ -587,7 +604,9 @@ namespace UnitTest
         AZ::NameDictionary::Create();
 
         // 3 threads per name effectively makes two readers and one writer (the first to run will write in the dictionary)
-        RunConcurrencyTest<ThreadCreatesOneName>(AZ_TRAIT_UNIT_TEST_NAME_COUNT, 3);
+        // Using AZ_TRAIT_UNIT_TEST_NAME_COUNT / 2 since the following line generates 3000 threads and that triggers ASan failures,
+        // likely because of https://devblogs.microsoft.com/oldnewthing/20050729-14/?p=34773
+        RunConcurrencyTest<ThreadCreatesOneName>(AZ_TRAIT_UNIT_TEST_NAME_COUNT / 2, 3);
```

Comment:
I think we should cut this trait down even more.
I would make the thread count 128 threads max.

Reply:
I dont disagree... just concerned what the intention was here... 
I will leave it up to the test author

Reply:
@santorac What are you thoughts on this, since your are the Test author.

1500 threads is still a lot and assuming a 1 Megabyte Stack per thread, that uses about 1.5 GiB of ram.
Considering that we consider the minimum [requirements](https://o3de.org/docs/welcome-guide/requirements/) for the engine 8GiB of CPU ram, that is still about 18% of the ram for the stack in these test..

Reply:
@santorac if you are ok with @lumberyard-employee-dm suggestion, I am happy to change it.
Note that the `3` there is a multiplier in the number of threads, so I would change `AZ_TRAIT_UNIT_TEST_NAME_COUNT` to something like `40`

Reply:
Keep in mind the 'AZ_TRAIT_UNIT_TEST_NAME_COUNT' is a PAL trait and was intended to be adjusted anyways per platform. I would think a better value for this should reflect how many threads the engine might use in normal circumstances.

Reply:
true, if @santorac is ok, I wont change `AZ_TRAIT_UNIT_TEST_NAME_COUNT` but just change the value that gets passed in this case.
Problem is that in this case threads are used to cause collisions on the use of Name variables and dictionary access. So the number is not really tied to how many threads should be running in the engine, its trying to force more race/contention conditions to make threading problems more likely to occur.

Reply:
Though my name is on this code, I'm not the original author (he left Amazon some time ago). I was responsible for finishing it up and making some final improvements. I concur that AZ_TRAIT_UNIT_TEST_NAME_COUNT is used for more than just driving thread counts, so I don't think we should change it for the sake of this situation; rather we should adjust the use of RunConcurrencyTest. 

I recommend creating a second trait AZ_TRAIT_UNIT_TEST_MAX_THREAD_COUNT = 2000 (or less if you want). Then modify the name loop in RunConcurrencyTest to go up to min(AZ_TRAIT_UNIT_TEST_MAX_THREAD_COUNT / threadsPerName, nameCount).

Reply:
will remove `AZ_TRAIT_UNIT_TEST_NAME_COUNT` because is just used in these tests and one other test where I can just set the value there
The trait is also in the wrong place, it was added to AzTest when the tests are in AzCore.Tests